contracts/interfaces/IAxelarGmpWithTokenGateway.sol

```diff
@@ -0,0 +1,118 @@
+    error BurnFailed(string symbol);
+    error MintFailed(string symbol);
+    error InvalidSetMintLimitsParams();
+    error ExceedMintLimit(string symbol);
+
+    /**********\
+    |* Events *|
+    \**********/
+
+    event TokenSent(
+        address indexed sender,
+        string destinationChain,
+        string destinationAddress,
+        string symbol,
+        uint256 amount
+    );
+
+    event ContractCallWithToken(
+        address indexed sender,
+        string destinationChain,
+        string destinationContractAddress,
+        bytes32 indexed payloadHash,
+        bytes payload,
+        string symbol,
+        uint256 amount
+    );
+
+    event TokenDeployed(string symbol, address tokenAddresses);
+
+    event ContractCallApprovedWithMint(
+        bytes32 indexed commandId,
+        string sourceChain,
+        string sourceAddress,
+        address indexed contractAddress,
+        bytes32 indexed payloadHash,
+        string symbol,
+        uint256 amount,
+        bytes32 sourceTxHash,
+        uint256 sourceEventIndex
+    );
+
+    event TokenMintLimitUpdated(string symbol, uint256 limit);
+
+    /********************\
+    |* Public Functions *|
+    \********************/
+
+    function sendToken(
+        string calldata destinationChain,
+        string calldata destinationAddress,
+        string calldata symbol,
+        uint256 amount
+    ) external;
+
+    function callContractWithToken(
+        string calldata destinationChain,
+        string calldata contractAddress,
+        bytes calldata payload,
+        string calldata symbol,
+        uint256 amount
+    ) external;
+
+    function isContractCallAndMintApproved(
+        bytes32 commandId,
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        address contractAddress,
+        bytes32 payloadHash,
+        string calldata symbol,
+        uint256 amount
+    ) external view returns (bool);
+
+    function validateContractCallAndMint(
+        bytes32 commandId,
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes32 payloadHash,
+        string calldata symbol,
+        uint256 amount
+    ) external returns (bool);
+
+    /***********\
+    |* Getters *|
+    \***********/
+
+    function tokenDeployer() external view returns (address);
+
+    function tokenMintLimit(string memory symbol) external view returns (uint256);
+
+    function tokenMintAmount(string memory symbol) external view returns (uint256);
+
+    function allTokensFrozen() external view returns (bool);
+
+    function tokenAddresses(string memory symbol) external view returns (address);
+
+    function tokenFrozen(string memory symbol) external view returns (bool);
+
+    /************************\
+    |* Governance Functions *|
+    \************************/
+
+    function setTokenMintLimits(string[] calldata symbols, uint256[] calldata limits) external;
```

Comment:
this should only include the GMP API, send, call contract, validate, tokenAddresses etc. Mint limit, tokenDeployer etc. are technical details of a specific gateway implementation